src/scripts/main.js

```diff
@@ -1,3 +1,244 @@
 'use strict';
 
-// write your code here
+const startBtn = document.querySelector('.button');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+const score = document.querySelector('.game-score');
+const gameScore = document.querySelector('.game-score');
+const rows = document.querySelectorAll('.field-row');
+
+let arrOfNumbers = [
+  ['', '', '', ''],
+  ['', '', '', ''],
+  ['', '', '', ''],
+  ['', '', '', ''],
+];
+let moves = 0;
+let maxNum = 0;
+
+score.innerText = maxNum;
+
+startBtn.addEventListener('click', (e) => {
+  e.target.classList.toggle('restart');
+
+  if (e.target.textContent === 'Start') {
+    e.target.textContent = 'Restart';
+    maxNum = 0;
+
+    arrOfNumbers = [
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+    ];
+    randomNum();
+    paint();
+  } else {
+    e.target.textContent = 'Start';
+  }
+});
+
+if (maxNum >= 2048) {
```

Comment:
The check for maxNum >= 2048 should be inside the game loop or a function that is called during the game loop, not at the top level of the script.